lib/list/getCSVData.js

```diff
@@ -0,0 +1,110 @@
+var _ = require('lodash');
+var listToArray = require('list-to-array');
+
+/**
+ * Applies option field transforms to get the CSV value for a field
+ */
+
+function transformFieldValue (field, item, options) {
+	var transform = typeof field.options.toCSV === 'string'
+		? listToArray(field.options.toCSV)
+		: field.options.toCSV;
+	if (typeof transform === 'function') {
+		return transform.call(item, field, options);
+	}
+	if (Array.isArray(transform)) {
+		var value = item.get(field.path);
+		if (transform.length === 1) {
+			return value[transform[0]];
+		} else {
+			return _.pick(value, transform);
+		}
+	}
+	return field.format(item);
+}
+
+/**
+ * Gets the data from an Item ready to be serialised to CSV for download
+ */
+
+function getCSVData (item, options) {
+	if (!options) {
+		options = {};
+	}
+	options.fields;
+	if (options.fields === undefined) {
+		options.fields = Object.keys(this.options.fields);
+	}
+	var data = {
+		id: String(item.id),
+	};
+	if (this.autokey) {
+		data[this.autokey.path] = item.get(this.autokey.path);
+	}
+	if (options.fields) {
+		if (typeof options.fields === 'string') {
+			options.fields = listToArray(options.fields);
+		}
+		if (!Array.isArray(options.fields)) {
+			throw new Error('List.getCSV: options.fields must be undefined, a string, or an array.');
+		}
+		options.fields.forEach(function (path) {
+			var field = this.fields[path];
+			if (field) {
```

Comment:
Can we do early returns here instead of this horrendously long `if/else`?

```JS
if (!fields) {
  return data[path] = item.get(path);
}

// For all fields except for expanded relationship fields transform the value 
if (!field.type === 'relationship' & options.expandRelationshipFields) {
  return data[path] = transformFieldValue(field, item, options);
}

var expanded = field.getExpandedData(item);

/* … */
```

That's a lot better to read imo